sklearnex/preview/ensemble/forest.py

```diff
@@ -0,0 +1,1096 @@
+            if not isinstance(self.max_leaf_nodes, numbers.Integral):
+                raise ValueError(
+                    "max_leaf_nodes must be integral number but was "
+                    "%r" %
+                    self.max_leaf_nodes)
+            if self.max_leaf_nodes < 2:
+                raise ValueError(
+                    ("max_leaf_nodes {0} must be either None "
+                     "or larger than 1").format(
+                        self.max_leaf_nodes))
+        if isinstance(self.max_bins, numbers.Integral):
+            if not 2 <= self.max_bins:
+                raise ValueError("max_bins must be at least 2, got %s"
+                                 % self.max_bins)
+        else:
+            raise ValueError("max_bins must be integral number but was "
+                             "%r" % self.max_bins)
+        if isinstance(self.min_bin_size, numbers.Integral):
+            if not 1 <= self.min_bin_size:
+                raise ValueError("min_bin_size must be at least 1, got %s"
+                                 % self.min_bin_size)
+        else:
+            raise ValueError("min_bin_size must be integral number but was "
+                             "%r" % self.min_bin_size)
+
+    def check_sample_weight(self, sample_weight, X, dtype=None):
+        n_samples = _num_samples(X)
+
+        if dtype is not None and dtype not in [np.float32, np.float64]:
+            dtype = np.float64
+
+        if sample_weight is None:
+            sample_weight = np.ones(n_samples, dtype=dtype)
+        elif isinstance(sample_weight, numbers.Number):
+            sample_weight = np.full(n_samples, sample_weight, dtype=dtype)
+        else:
+            if dtype is None:
+                dtype = [np.float64, np.float32]
+            sample_weight = check_array(
+                sample_weight,
+                accept_sparse=False,
+                ensure_2d=False,
+                dtype=dtype,
+                order="C")
+            if sample_weight.ndim != 1:
+                raise ValueError("Sample weights must be 1D array or scalar")
+
+            if sample_weight.shape != (n_samples,):
+                raise ValueError("sample_weight.shape == {}, expected {}!"
+                                 .format(sample_weight.shape, (n_samples,)))
+        return sample_weight
+
+
+class RandomForestClassifier(sklearn_RandomForestClassifier, BaseRandomForest):
+    __doc__ = sklearn_RandomForestClassifier.__doc__
+
+    if sklearn_check_version('1.2'):
+        _parameter_constraints: dict = {
+            **sklearn_RandomForestClassifier._parameter_constraints,
+            "max_bins": [
+                Interval(
+                    numbers.Integral,
+                    2,
+                    None,
+                    closed="left")],
+            "min_bin_size": [
+                Interval(
+                    numbers.Integral,
+                    1,
+                    None,
+                    closed="left")]}

```

Comment:
Is it produced by auto linter? This code will be more readable without so many line breaks

Reply:
Fixed